priv/repo/migrations/20220418150726_add_publisher_to_projects_and_products.exs

```diff
@@ -0,0 +1,52 @@
+defmodule Oli.Repo.Migrations.AddPublisherToProjectsAndProducts do
+  use Ecto.Migration
+
+  import Ecto.Query, warn: false
+
+  alias Oli.Delivery.Sections.Section
+  alias Oli.Inventories
+  alias Oli.Inventories.Publisher
+  alias Oli.Repo
+
+  require Logger
+
+  def up do
+    default_publisher_attrs = %{
+      name: Inventories.default_publisher_name(),
+      email: "publisher@cmu.edu"
+    }
+
+    _publisher =
+      Publisher
+      |> struct(default_publisher_attrs)
+      |> Repo.insert!(on_conflict: :nothing)
+
+    %Publisher{id: publisher_id} = Repo.get_by!(Publisher, default_publisher_attrs)
+
+    flush()
+
+    alter table(:sections) do
+      add :publisher_id, references(:publishers)
+    end
+
+    alter table(:projects) do
+      add :publisher_id, references(:publishers), null: false, default: publisher_id
+    end
+
+    flush()
+
+    # Set default publisher for products
+    from(s in Section, where: s.type == :blueprint)
```

Comment:
I would think that we need a similar UPDATE query to set the publisher_id for all existing projects.

Reply:
Actually we're doing that above by setting the default value when adding the `publisher_id` to projects.

Reply:
It affects only existing records. We have the `null: false` constraint in the db and also a schema validation that doesn't allow new projects without a publisher.

Reply:
Ok, great.  I was unaware that `add :publisher_id, references(:publishers), null: false, default: publisher_id` in a migration would actually set the values in the existing rows.